flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceEnumeratorStateSerializer.java

```diff
@@ -54,12 +44,9 @@ public int getVersion() {
         try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
                 DataOutputStream out = new DataOutputStream(baos)) {
             out.writeInt(enumState.getCurrentSourceIndex());
-            SimpleVersionedSerializer<Object> serializer =
-                    serializerOf(enumState.getCurrentSourceIndex());
-            out.writeInt(serializer.getVersion());
-            byte[] enumStateBytes = serializer.serialize(enumState.getWrappedState());
-            out.writeInt(enumStateBytes.length);
-            out.write(enumStateBytes);
+            out.writeInt(enumState.wrappedStateSerializerVersion());
+            out.writeInt(enumState.getWrappedState().length);
```

Comment:
integer would limit the state size to 2 GB. not sure if we need to worry about it or not. It can happen if the historical storage (like HDFS or Iceberg) have many files/splits for the booststrap scan.

Reply:
Each Iceberg split contains data files, delete files (for upsert), schema string. Each data file also contains stats for every column. if the table is wide (many columns), each split may go over 10 KB

Reply:
This discussion is probably outside the scope of this PR

Reply:
Please note that the limit would not be the integer to represent the size but rather the byte[] array that cannot go beyond that. We do not have 64 bit array https://www.nayuki.io/page/large-arrays-proposal-for-java

Reply:
yes. understood it is the choice of bye[], which is then a limitation of SimpleVersionedSerializer's API

Reply:
I also wonder if we would hit other issues with such large state serialized in the coordinator? Can IcebergSource limit the number of splits it keeps in the checkpoint and only add more once some have been processed?

Reply:
@tweise I have thought about adding the optimization (of limiting the number of splits) for streaming read in the future. for bounded job, we can't. on the other hand, bounded job may not need to have checkpoint enabled